crypto/pkcs7/pkcs7_x509.c

```diff
@@ -370,6 +377,93 @@ static int write_signer_info(CBB *out, const void *arg) {
   return ret;
 }
 
+static int pkcs7_add_signature(PKCS7 *p7, X509 *x509, EVP_PKEY *pkey) {
+  // OpenSSL's docs say that this defaults to SHA1, but appears to actually
+  // default to SHA256 in 1.1.x and 3.x for RSA, DSA, and EC(DSA).
+  // https://linux.die.net/man/3/pkcs7_sign
+  // https://github.com/openssl/openssl/blob/79c98fc6ccab49f02528e06cc046ac61f841a753/crypto/rsa/rsa_ameth.c#L438
+  const EVP_MD *digest = EVP_sha256();
+  PKCS7_SIGNER_INFO *si = NULL;
```

Comment:
Seems like this is similar to the issue I'm trying to fix for this PR in the OCSP layer: https://github.com/aws/aws-lc/pull/2038